packages/licensing/src/class-licensing.php

```diff
@@ -0,0 +1,219 @@
+<?php
+/**
+ * A Terms of Service class for Jetpack.
+ *
+ * @package automattic/jetpack-licensing
+ */
+
+namespace Automattic\Jetpack;
+
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Jetpack;
+use Jetpack_IXR_ClientMulticall;
+use Jetpack_Options;
+use WP_Error;
+
+/**
+ * Class Licensing.
+ * Helper class that is responsible for attaching licenses to the current site.
+ *
+ * @since ??
+ */
+class Licensing {
+	/**
+	 * Name of the WordPress option that holds all known Jetpack licenses.
+	 *
+	 * @const string
+	 */
+	const LICENSES_OPTION_NAME = 'jetpack_licenses';
+
+	/**
+	 * Name of the WordPress transient that holds the last license attaching error, if any.
+	 *
+	 * @const string
+	 */
+	const ERROR_TRANSIENT_NAME = 'jetpack_licenses_error';
+
+	/**
+	 * Holds the singleton instance of this class.
+	 *
+	 * @var self
+	 */
+	protected static $instance = false;
+
+	/**
+	 * Singleton.
+	 *
+	 * @static
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initialize.
+	 *
+	 * @return void
+	 */
+	public function initialize() {
+		add_action( 'update_option_' . self::LICENSES_OPTION_NAME, array( $this, 'attach_stored_licenses' ) );
+		add_action( 'jetpack_authorize_ending_authorized', array( $this, 'attach_stored_licenses_on_connection' ) );
+		add_action( 'load-toplevel_page_jetpack', array( $this, 'surface_error' ) );
+	}
+
+	/**
+	 * Get Jetpack connection manager instance.
+	 *
+	 * @return Connection_Manager
+	 */
+	protected function connection() {
+		return new Connection_Manager();
```

Comment:
I would suggest storing the instance in a property after it's first initialized, instead of creating new object every time:
```php
if ( null === $this->connection ) {
    $this->connection = new Connection_Manager();
}

return $this->connection;
```